### PR TITLE
Fixed inserting Unix EOL

### PR DESCRIPTION
**Pull request by aecz**
- Added eolChar parameter to wiring.generateBlock function
- Replaced fixed Unix EOL with eolChar variable
- Assigned eolChar value depending on HTML string
- Added Unix EOL to tests using generateBlock function
- Added tests with an HTML string using Windows and Unix EOL
- Added test using a real HTML file as input

Based on NodeJS code, there is only 2 possible EOL:
https://github.com/joyent/node/blob/master/lib/os.js#L66

I also noticed that tests using fixtures uses **textEqual** instead of **equal**.
This could be changed if the file **basic_html.html** is used as input instead of a string. Also **trim** function could be removed.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -159,17 +160,23 @@ wiring.appendFiles = function appendFiles(htmlOrOptions, fileType, optimizedPath
 
   attrs = this.attributes(attrs);
 
+  if (html.indexOf('\r\n') > 0) {
```
Comment (sindresorhus) [inline index.js]: Use https://github.com/sindresorhus/detect-newline/ for a more reliable detection.

File: index.js [hunk 2]
```diff
@@ -29,7 +30,9 @@ var wiring = module.exports;
  */
 
 wiring.domUpdate = function domUpdate(html, tagName, content, mode) {
-  var $ = cheerio.load(html, { decodeEntities: false });
+  var $ = cheerio.load(html, {
+    decodeEntities: false
+  });
```
Comment (arthurvr) [inline index.js]: Why did you change this?

Comment (aecz) [inline index.js]: It is only a formatting change as you can see done by BeautifyJS.
Which formatter are you using for Javascript ?

Comment (arthurvr) [inline index.js]: In this repo, none. Even if it was automated let's not do random changes in unrelated commits ;)

**Comment by aecz** [comment]
@arthurvr I have removed the formatting change.

**Comment by SBoudrias** [comment]
Merged on master! Thanks :)

**Comment by SBoudrias** [comment]
Merged on master! Thanks :)
